### PR TITLE
Resolve compiler messages in `ScenarioTests`, use `FluentAssertions`

### PR DESCRIPTION
**Pull request by premun**
This PR improves the work with the ScenarioTests project code:
- Ran analyzers and resolved compiler messages
- Bumped testing libraries
- Stopped using `NUnit.Assert` and started using `FluentAssertions` for better test failure messages
- Formatted code to match the rest of the repo
- Enabled nullability for some classes

There should be no functional changes, just re-formatting and adding nullability. Just errors from failed assertions should get better.

https://github.com/dotnet/arcade-services/issues/3031

This build verifies everything: https://dev.azure.com/dnceng/internal/_build/results?buildId=2340341&view=results

### Release Note Category
- [ ] Feature changes/additions 
- [ ] Bug fixes
- [x] Internal Infrastructure Improvements

### Release Note Description
Not release notes worthy

### CODE REVIEW

**Comment by premun** [comment]
This build verifies everything: https://dev.azure.com/dnceng/internal/_build/results?buildId=2340310

**Review by dkurepa — APPROVED** [review]

File: test/Microsoft.DotNet.DarcLib.Tests/GitHubClientTests.cs [hunk 1]
```diff
@@ -398,9 +398,9 @@ private async Task<IList<Review>> GetLatestReviewsForPullRequestWrapperAsync(Dic
 
         // Use Moq to put the return value 
         OctoKitPullRequestReviewsClient.Setup(x => x.GetAll(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<int>()))
-            .Callback<string, string, int>((string x, string y, int z) =>
+            .Callback((string x, string y, int z) =>
             {
-                Tuple<string, string, int> theKey = new Tuple<string, string, int>(x, y, z);
+                var theKey = new Tuple<string, string, int>(x, y, z);
```
Comment (andriipatsula) [inline test/Microsoft.DotNet.DarcLib.Tests/GitHubClientTests.cs:403]: I believe having a type specified has more pros than cons.

**Review by andriipatsula — APPROVED** [review]

File: test/Microsoft.DotNet.DarcLib.Tests/GitHubClientTests.cs (hunk 1 above)
Comment (premun) [reply test/Microsoft.DotNet.DarcLib.Tests/GitHubClientTests.cs:403]: But the type is still there, just not twice in the same row?

Comment (premun) [reply test/Microsoft.DotNet.DarcLib.Tests/GitHubClientTests.cs:403]: This only changes the "it's obvious" cases

**Pull request merged**